proxy.php

```diff
@@ -272,16 +273,20 @@ function getHttpContentAndStatus($url, $timeout, $user_agent)
 
     // if there's POST data, send our proxy request as a POST
     if (!empty($_POST)) {
-        $postBody = http_build_query($_POST);
+        $postBody = file_get_contents("php://input");
 
         $stream_options['http']['method'] = 'POST';
         $stream_options['http']['header'][] = "Content-type: application/x-www-form-urlencoded";
         $stream_options['http']['header'][] = "Content-Length: " . strlen($postBody);
         $stream_options['http']['content'] = $postBody;
+        
+        if(!empty($http_ip_forward_header)) {
+            $visitIp = getVisitIp();
+            $stream_options['http']['header'][] = "$http_ip_forward_header: $visitIp";
+        }
     }
 
     if($useFopen) {
-        $stream_options['http']['header'] = implode("\r\n", $header);
```

Comment:
just checking is this here done on purpose?

Reply:
Yes. When using the implode method by concatenating all headers in one string it didn't work.
Starting from ca. php 5.2 ```stream_context_create``` accepts an array for multiple headers to be set.
See https://www.php.net/manual/en/context.http.php 
Therefore the implode line could be omitted.

It seems that the implode method is not enough as the header string needs an additional ```"\r\n"```.

https://stackoverflow.com/questions/17467601/php-http-request-stream-context-create
but why care if ```stream_context_create``` does it the right way if an array of headers is given.

Reply:
I've re-visited the php error log backwards before I made this change and found the following entry:
```
PHP Notice:  file_get_contents(): Content-type not specified assuming application/x-www-form-urlencoded in /matomo/proxy.php on line 288
```
which also indicates that the headers were not sent correctly before.

Reply:
Mmpf: hold it. It is OK as it is now, as the header is set here: https://github.com/matomo-org/tracker-proxy/blob/master/proxy.php#L257
So no need to change something.